aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -417,21 +456,44 @@ public Builder subscriptionsAutoReconnect( boolean subscriptionsAutoReconnect) {
          * Default value is 5 minutes. Note that this limit is for execution time - the mutation can wait in the queue for its turn to
          * be processed independent of this limit.
          * @param mutationQueueExecutionTimeout the max execution time allowed.
-         * @return
+         *
+         * @return the builder object
          */
         public Builder mutationQueueExecutionTimeout(long mutationQueueExecutionTimeout) {
             mMutationQueueExecutionTimeout = mutationQueueExecutionTimeout;
             return this;
         }
 
-        public AWSAppSyncClient build() {
-            if (mNormalizedCacheFactory == null) {
-                AppSyncSqlHelper appSyncSqlHelper = AppSyncSqlHelper.create(mContext, defaultSqlStoreName);
+        /**
+         * Flag if true will use the ClientDatabasePrefix passed in through the
+         * awsconfiguration.json or from the #clientDatabasePrefix builder.
+         *
+         * @param useClientDatabasePrefix use a prefix for the client databases.
+         * @return the builder object.
+         */
+        public Builder useClientDatabasePrefix(@NonNull boolean useClientDatabasePrefix) {
```

Comment:
Consider calling this "useClientPrefix" or "useClientName" instead. No need to leak the idea that the clientName is used to create a separate database

Reply:
For being explicit and transparent, we decided to use "useClientDatabasePrefix". We are not introducing the concept of a client name or client identifier.

Reply:
sounds good.